src/components/app/app.tsx

```diff
@@ -1,18 +1,36 @@
 import React from 'react';
-import MainPage from '../main-page/main-page';
+import { Switch, Route, BrowserRouter } from 'react-router-dom';
+import { AppRoute } from '../../const';
+import MainPage from '../pages/main';
+import FavoritesPage from '../pages/favorites';
+import RoomPage from '../pages/room';
+import SignInPage from '../pages/sign-in';
+import NotFoundPage from '../pages/not-found';
+import { PageMainProps } from '../pages/types';
 
-interface Props {
-  cardCount: number,
-  placesToStay: number,
-}
-
-function App({ cardCount, placesToStay }: Props): React.ReactElement {
+export default function App({ cardCount, placesToStay }: PageMainProps): React.ReactElement {
```

Comment:
```suggestion
export default function App(p: PageMainProps): React.ReactElement {
  const { cardCount, placesToStay } = p;
```
Такая нотация подсказывает читателю, что компонент обещает не изменять значения свойств, полученных от родителя